docs/v1/api.md

```diff
@@ -0,0 +1,335 @@
+# Service Catalog API
+
+This document details a standard API, including a schema for standard
+Kubernetes-native resources, to:
+
+- Formalize an API around the service broker concept as a first-class Kubernetes
+  (or Kubernetes-like) API for all to develop against.
+- Follow established patterns in the Kubernetes core API for globally-available
+  and namespaced resources
+- Facilitate the development of Kubernetes-like controllers in support of the
+  API, without undermining competition or choice
+- Ensure a smooth, consistent UX for users of the API and implementors of the
+  controllers
+
+# API Precedents in Kubernetes
+
+Broadly speaking, the system detailed herein can list services and allow
+consumers to provision, bind, unbind and deprovision them.
+
+We've drawn inspiration from the
+[Persistent Volumes](http://kubernetes.io/docs/user-guide/persistent-volumes/)
+API and the [Ingress Controllers](http://kubernetes.io/docs/user-guide/ingress/)
+system. The remainder of this section explains these inspirations in further
+detail.
+
+Additionally, we've adhered, wherever possible, to Kubernetes's declarative
+API. That is, the resources we've defined herein describe the _desired end
+state_ of the applicable systems. Kubernetes and the service catalog controller
+implementations (see below) are responsible for fulfilling that end state.
+
+## Persistent Volumes
+
+First, this problem space is similar to that of persistent volumes. We model
+the data structures herein after those in persistent volumes primarily because
+we believe that consistency is a virtue. More importantly, the community
```

Comment:
Consistency is also important to giving users an experience that feels designed instead of organically grown / ad-hoc.